### DISCUSSION TITLE
How to use an Import Node for PCAPs

### DISCUSSION THREAD

**Discussion by Sazerak** [Security-Onion-Solutions/securityonion#7236] [Unsupported Versions] [answered]
I've been using an import node recently for PCAPs and realized I probably wasn't using it right. At the end of the day, I would use so-elastic-clear to clear my data to load up another PCAP in the morning and only see the new alerts/hunt data, etc.. but today I needed to re-ingest an earlier PCAP and SO wouldn't do it because it identified that it had already been imported.

So that got me asking myself, if the file is still there but I can't see it, maybe so-elastic-clear isn't the command I should be using to reset my import node at the end of the day. I have a look through the documentation and the cheat sheet, but can't seem to figure out which commands I should be using and when. I'm seeing other commands related to PCAPs like so-pcap-start but can't find any reference as to what they do or how they're used.

**Comment by Sazerak** [comment]
I think I partially answered my question. Looks like what I'm probably looking for is so-nsm-clear, but I also found that the imported PCAPs are in /nsm/import which so-nsm-clear doesn't seem to remove. Can someone confirm? Also, even if I have my answer, would it be reasonable to ask for a complete guide to all so commands, something like a man page that anyone can quickly refer to to get the details on a specific command? I can only seem to find information on the more common commands.

**Reply by Sazerak** [reply]
Actually it looks like so-nsm-clear doesn't work on an import node because so-zeek isn't installed.

Edit: ok so this is strange. When I reboot my VM, so-zeek shows it's enabled (OK) for the first little while, so during that time I can run an so-nsm-clear. So I tried running so-zeek-start when zeek was showing Disabled and it started up fine. That allowed me to run so-nsm-clear.

But is this really how it's supposed to be done? Seems like there must be an easier way than this.

Edit 2: OK, so so-nsm-clear did what it's supposed to, but it doesn't remove the things I'm actually looking to remove which is the contents of /nsm/import and /nsm/pcaptmp. Is there another command possibly to clear these or do we just do it manually?

**Comment by dougburks** [comment, marked as answer]
I've added the following to https://docs.securityonion.net/en/2.3/so-import-pcap.html:
> so-import-pcap calculates the MD5 hash of the imported pcap and creates a directory in /nsm/import/ for that hash. This is where so-import-pcap stores the alerts and logs generated by the traffic in the pcap. If you try to import that same pcap again, it will tell you that it has already imported that pcap. If for some reason you really do need to import that pcap again, you can remove that pcap’s directory in /nsm/import/ and then try again.

Here's what I do for our [Quick Malware Analysis blog posts](https://blog.securityonion.net/search/label/quick%20malware%20analysis):
- create a VM and install the latest version of Security Onion in IMPORT mode
- once everything is configured the way I want it, create a VM snapshot

Then every time I want to import a pcap:
- revert to VM snapshot
- run so-import-pcap

That way I don't have to manually delete anything and everything is in a known good state.

**Reply by Sazerak** [reply]
Thanks for the reply. I had noticed that when I tried importing again, it gave me the hash so it's good to know I can just find it that way and reimport it. And yes you're right that using snapshots is probably best and I'll do it that way from here on.
